# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/RtpPacketQueue.h
# Commit: 3fb02a012faa
# Full Hash: 3fb02a012faadb59c6a4ca42bfb50bcbb1037d93
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Route packets with MediaEvents, and clean up conduits' cross-thread syncing. r=bwc,ng
#   
#   This patch solves some issues with the SSRC switching that can be triggered by
#   both signaling and incoming packets, on different threads. But mainly, because
#   that setup is inherently complex, this patch greatly simplifies the SSRC
# ==============================================================================

diff -r 336f58a54bd8 -r 3fb02a012faa dom/media/webrtc/libwebrtcglue/RtpPacketQueue.h
--- a/dom/media/webrtc/libwebrtcglue/RtpPacketQueue.h	Wed Aug 25 20:28:12 2021 +0200
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,46 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef RtpPacketQueue_h
-#define RtpPacketQueue_h
-
-#include "MediaConduitInterface.h"
-#include "nsTArray.h"
-
-namespace mozilla {
-
-class RtpPacketQueue {
- public:
-  void Clear() {
-    mQueuedPackets.Clear();
-    mQueueActive = false;
-  }
-
-  void DequeueAll(MediaSessionConduit* conduit) {
-    // SSRC is set; insert queued packets
-    for (auto& packet : mQueuedPackets) {
-      conduit->DeliverPacket(std::move(packet),
-                             MediaSessionConduit::PacketType::RTP);
-    }
-    mQueuedPackets.Clear();
-    mQueueActive = false;
-  }
-
-  void Enqueue(rtc::CopyOnWriteBuffer packet) {
-    mQueuedPackets.AppendElement(std::move(packet));
-    mQueueActive = true;
-  }
-
-  bool IsQueueActive() { return mQueueActive; }
-
- private:
-  bool mQueueActive = false;
-  nsTArray<rtc::CopyOnWriteBuffer> mQueuedPackets;
-};
-
-}  // namespace mozilla
-
-#endif  // RtpPacketQueue_h